src/chordparser/editors/chord_roman_converter.py

```diff
@@ -83,11 +83,12 @@ def to_roman(self, chord, scale_key):
                 UserWarning
             )
             chord = self._CE.change_chord(chord, quality="Maj", inplace=False)
+            
         if isinstance(scale_key, Scale):
-            scale_root = scale_key.key.root
+            scale = scale_key
         else:
-            scale_root = scale_key.root
-        scale = self._SE.create_scale(scale_root, "major")
+            scale = self._SE.create_scale(scale_key.root, scale_key.mode)
```

Comment:
Hm actually, is there a need to differentiate natural/harmonic/melodic minor? I don't know how those interact with the Roman numeral analysis but if there's meaning to that then maybe we can include the key's submode too.